File: Linear Search.c

//Implementation of a Linear Search using Arrays
#include <stdio.h>
int main()
{
	int n,s;//Variablesd for size of array and search element
	printf("Enter the number of elements in the array\n");
	scanf("%d",&n);
	int a[n];
	printf("Enter the array elements\n");
	int i;
	for(i=0;i<n;i++)
		scanf("%d",&a[i]);//Inputting the Array
	printf("Enter the value to be searched\n");
		scanf("%d",&s);//Value to be Searched
	int c=-1;//default assumption element not present
	for(i=0;i<n;i++)
	{
		if(a[i]==s)
		{
			c=i;
			printf("The value %d is present at the position %d in the array\n",s,(c+1));
			//for multiple instances in the array
		}
	}
	if(c==-1)
		printf("Element not found\n");
	printf("\n=>The Array-:\n");//Printing the array for reference
	for(i=0;i<n;i++)
		printf("\t%d",a[i]);
	return 1;
}

File: Linear_Search.java
import java.util.*;
public class Linear_Search
{
    int size;
    int a[];
    Linear_Search(int n)
    {
        size=n;
        a=new int[n];
    }
    void input()
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the elements");
        for(int i=0;i<size;i++)
            a[i]=sc.nextInt();
    }
    void search(int n)
    {
        int flag=0;
        for(int i=0;i<size;i++)
        {
            if(a[i]==n)
            {
                System.out.println("Found at "+(i+1));
                flag++;
            }
        }
        if(flag==0)
            System.out.println("Not Found");
    }
    void display()
    {
        System.out.println("\f");
        System.out.println(Arrays.toString(a));
    }
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter size of array");
        Linear_Search obj=new Linear_Search(sc.nextInt());
        obj.input();
        System.out.println("Enter Element");
        int c=sc.nextInt();
        obj.display();
        obj.search(c);
    }
}
